src/test/java/org/jboss/eap/qa/jarest/DistributionTestCase.java

```diff
@@ -79,17 +96,19 @@ void checkExpectedMRv9JARs() throws IOException {
     @Test
     void atMostMRv9JARsPresent() throws IOException {
         SoftAssertions softly = new SoftAssertions();
-        distributionDir.jarsAsZipFiles()
-                .forEach(zipFile -> {
+        distributionDir.jars()
+                .filter(jar -> jar.hasOverlayDirectory())
+                .forEach(jar -> {
                             softly.assertThat(
-                                    zipFile.stream()
-                                            .filter(entry -> entry.getName().contains("META-INF/versions"))
-                                            .filter(entry -> !entry.getName().contains("META-INF/versions/9"))
-                                            .filter(entry -> !entry.getName().equals("META-INF/versions/"))
-                                            .peek(System.out::println)
-                                            .count())
-                                    .as("Artifact %s is expected to be at most versions/9 Multi-Release JAR", zipFile.getName())
-                                    .isEqualTo(0);
+                                    jar.hasOverlayDirectory(9)
+                                            && ! jar.hasOverlayDirectory(10)
+                                            && ! jar.hasOverlayDirectory(11)
+                                            && ! jar.hasOverlayDirectory(12)
+                                            && ! jar.hasOverlayDirectory(13)
+                                            && ! jar.hasOverlayDirectory(14)
+                                    )
```

Comment:
This seems a bit ugly to me. Perhaps a method checking that the `jar` has overlay directories only up to version x would be better.

Reply:
Alright, I'm fine with this if it is one time method.